sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java

```diff
@@ -52,25 +48,31 @@ public ChooseSourceCaseSteps(
         "^I open the first found result in the CHOOSE SOURCE window$",
         () -> {
           webDriverHelpers.clickOnWebElementBySelector(SOURCE_CASE_WINDOW_FIRST_RESULT_OPTION);
-          webDriverHelpers.clickWhileOtherButtonIsDisplayed(
-              SOURCE_CASE_WINDOW_CONFIRM_BUTTON, CHANGE_CASE_BUTTON);
+          webDriverHelpers.waitUntilElementIsVisibleAndClickable(SOURCE_CASE_WINDOW_CONFIRM_BUTTON);
+          webDriverHelpers.clickOnWebElementBySelector(SOURCE_CASE_WINDOW_CONFIRM_BUTTON);
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
+              SUCCESSFUL_CASE_CHANGE_POPUP);
         });
 
     When(
         "I click on the CHOOSE SOURCE CASE button",
         () ->
             webDriverHelpers.clickWhileOtherButtonIsDisplayed(
-                CHOOSE_SOURCE_CASE_BUTTON, DISCARD_POPUP_YES_BUTTON));
+                CHOOSE_SOURCE_CASE_BUTTON, POPUP_YES_BUTTON));
 
     When(
         "I click yes on the DISCARD UNSAVED CHANGES popup",
```

Comment:
also here, can we add also the location of where this action happens?

Reply:
reworked